api/provider/aws/load_balancer_delete.go

```diff
@@ -28,23 +28,23 @@ func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
 	}
 
 	// Check for eventually consistency
-	waitUntilLBDeletedFN := func() (shouldRetry bool, err error) {
+	var err error
+	waitUntilLBDeletedFN := func() (shouldRetry bool) {
 		loadBalancerName := fqLoadBalancerID
 		if _, err = describeLoadBalancer(l.AWS.ELB, l.AWS.ALB, loadBalancerName); err != nil {
 			if err, ok := err.(*errors.ServerError); ok && err.Code == errors.LoadBalancerDoesNotExist {
-				return false, nil
+				return false
 			}
 
-			return false, err
+			return false
 		}
 
 		log.Printf("[DEBUG] Load Balancer not deleted, will retry lookup")
-		return true, nil
+		err = errors.New(errors.EventualConsistencyError, err)
```

Comment:
again, err is nil at this point so you need to create one to send to `errors.New` or use `errors.Newf`